### ISSUE TITLE
[DEV] rinv does not handle the firmware output well when multiple images are on the BMC

### ISSUE THREAD

**Issue by whowutwut** [xcat2/xcat-core#3274]
When the BMC contains multiple firmware images, the output for `rinv firm` does not look pretty. 

```
[root@fs2vm110 ~]# rinv p9euh02 firm
p9euh02: BMC SOFTWARE: v1.99.6-130-g35cfa84 (Active)
p9euh02: HOST SOFTWARE: IBM-witherspoon-ibm-OP9_v1.16_1.24 (Ready)
p9euh02: HOST SOFTWARE: IBM-witherspoon-ibm-OP9_v1.17_1.10 (Ready)
p9euh02: HOST SOFTWARE: IBM-witherspoon-ibm-OP9_v1.17_1.20 (Ready)
p9euh02: HOST SOFTWARE: -- additional info: buildroot-2017.2.2-7-g23118ce
p9euh02: HOST SOFTWARE: -- additional info: buildroot-2017.2.2-7-g23118ce
p9euh02: HOST SOFTWARE: -- additional info: buildroot-2017.2.2-7-g23118ce
p9euh02: HOST SOFTWARE: -- additional info: capp-ucode-9c73e9f
p9euh02: HOST SOFTWARE: -- additional info: capp-ucode-9c73e9f
p9euh02: HOST SOFTWARE: -- additional info: capp-ucode-9c73e9f
p9euh02: HOST SOFTWARE: -- additional info: hostboot-binaries-bc600e6
p9euh02: HOST SOFTWARE: -- additional info: hostboot-binaries-d8f4ab7
p9euh02: HOST SOFTWARE: -- additional info: hostboot-binaries-d8f4ab7
p9euh02: HOST SOFTWARE: -- additional info: hostboot-d2273f
p9euh02: HOST SOFTWARE: -- additional info: hostboot-d2273f
p9euh02: HOST SOFTWARE: -- additional info: hostboot-dff36e4
p9euh02: HOST SOFTWARE: -- additional info: linux-4.11.1-openpower1-p77ac250
p9euh02: HOST SOFTWARE: -- additional info: linux-4.11.1-openpower1-pc11ec3
p9euh02: HOST SOFTWARE: -- additional info: linux-4.11.2-openpower1-p62f2cf
p9euh02: HOST SOFTWARE: -- additional info: machine-xml-96746e9
p9euh02: HOST SOFTWARE: -- additional info: machine-xml-e536539
p9euh02: HOST SOFTWARE: -- additional info: machine-xml-e536539
p9euh02: HOST SOFTWARE: -- additional info: occ-8a7df3
p9euh02: HOST SOFTWARE: -- additional info: occ-8a7df3
p9euh02: HOST SOFTWARE: -- additional info: occ-bc34ad5
p9euh02: HOST SOFTWARE: -- additional info: op-build-v1.16-83-ge67b53c-dirty
p9euh02: HOST SOFTWARE: -- additional info: op-build-v1.16-87-g5e69953-dirty
p9euh02: HOST SOFTWARE: -- additional info: op-build-v1.17-20-gbaac7a6-dirty
p9euh02: HOST SOFTWARE: -- additional info: petitboot-v1.4.2-p9717304
p9euh02: HOST SOFTWARE: -- additional info: petitboot-v1.4.2-pcb6424c
p9euh02: HOST SOFTWARE: -- additional info: petitboot-v1.4.2-pf0e80a2
p9euh02: HOST SOFTWARE: -- additional info: sbe-d770027
p9euh02: HOST SOFTWARE: -- additional info: sbe-d770027
p9euh02: HOST SOFTWARE: -- additional info: sbe-d770027
p9euh02: HOST SOFTWARE: -- additional info: skiboot-5.6.0-rc2
p9euh02: HOST SOFTWARE: -- additional info: skiboot-5b8834-p7542d9b
p9euh02: HOST SOFTWARE: -- additional info: skiboot-d673ecc
```

**Comment by whowutwut** [comment]
@gurevichmark Should we only display the active image and let rflash display the list?  Or just display all... and formatted better.

**Comment by gurevichmark** [comment]
I think for `rinv firm` we should just display currently active image and let `rflash --list` display all of them. I feel this will be more consistent with ipmi output.

**Comment by zet809** [comment]
I agree with @gurevichmark that only display all firmware information with `rflash` to keep consistent with what ipmi do.

**Comment by whowutwut** [comment]
Ok, sounds good.  Here's the planned sample output that we will display for rinv.. 
```
# rinv p9euh0[1-2] firm
p9euh01: BMC SOFTWARE[6584ef1]: v1.99.6-19-gab8de5b (Active)
p9euh01: HOST SOFTWARE[488449a2]: IBM-witherspoon-ibm-OP9_v1.17_1.20 (Active)
p9euh02: BMC SOFTWARE[5ea407c0]: v1.99.6-130-g35cfa84 (Active)
```

**Comment by robin2008** [comment]
Is it possible to make the output of firmware information similar as garrison and firestone, as they are OPAL too.
```
[root@stratton01 ~]# rinv frame5u39 firm
frame5u39: BMC Firmware: 2.13
frame5u39: System Firmware Product Version: IBM-firestone-ibm-OP8_v1.8_1.1
frame5u39: System Firmware Product Additional Info: hostboot-bc98d0b-74f74b1
frame5u39: System Firmware Product Additional Info: occ-0362706-opdirty-60bbdf1
frame5u39: System Firmware Product Additional Info: skiboot-5.1.16
frame5u39: System Firmware Product Additional Info: hostboot-binaries-43d5a59
frame5u39: System Firmware Product Additional Info: firestone-xml-e7b4fa2-8656aaa
frame5u39: System Firmware Product Additional Info: capp-ucode-105cb8f-opdirty
```
And maybe for the "Additional Info" for the System Firmware (HOST Software), we can have a flag for it to be shown or hidden.  Not sure if "-V" could be used for this purpose. Till now, I didn't see any difference w/ or w/o -V for `rinv foo firm`

**Comment by whowutwut** [comment]
Sure will look into this to sync up the output and create a pull request...

If we really want the support for verbose. I think another issue should be opened for that. "firm" is an option to rinv and we need to think about what verbose would provide with the rinv command in general. Seems like no options prints all. Them firm already reduces the subset. Then -V to add more. This needs some thought..

**Comment by whowutwut** [comment]
@robin2008  I looked into this more today.  I  had forgotten the reason it was formatted this way and it came back to me while I investigated this today.    In the ipmi version, we have code that is making calls and holding data separated and then printing it out.  

In the Rest code, we call a single command to get all the firmware levels and then sort it best we can.  To keep the data in the right order, we have to play with the output so that Version gets printed before Additional Info.   

I'll make some changes since I started improving this today, but I don't think we will match the OpenPower P8 machines.
